charts/argo-cd/README.md.gotmpl

```diff
@@ -91,10 +94,38 @@ Please use `kubectl` to upgrade CRDs manually from [crds](crds/) folder or via t
 ```bash
 kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=<appVersion>
 
-# Eg. version v2.3.3
-kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.3.3
+# Eg. version v2.4.0
+kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.4.0
 ```
 
+### 5.0.0
+
+This version introduces large refactoring aimed to consolidate all configuration sections in single place.
+
+General:
+- Added new `configs.params` section that replaces all `args` from Argo components. Please refer to [argocd-cmd-params-cm] for equivalents provided by `args`.
+- Log level and format can be now set globally or via `configs.params` per Argo CD component.
+- GPG keys, SSH keys and TLS certs now have consolidated configuration sections
+- Removed deprecated configuration keys without replacement
+  - configs.repositoryCredentials
+  - server.config.repositories
+
+ApplicationSet:
+- Renamed parameters to align with rest of the chart
+  - `replicaCount` to `replicas`
+  - `securityContext` to `containerSecurityContext`
+  - `extraEnv` to `env`
+  - `extraEnvFrom` to `envFrom`
```

Comment:
Maybe it's time to analyze the other charts and consolidate the naming. I merged one PR for rollouts with extraEnv couple of hours ago :(

Reply:
Agree one same naming convention. All original components (server, reposerver and controller) were using env and envFrom so I tried to stick with that. If you want to change these to extraEnv it is also fine. Imo I believe env will be used less often because of params-cm.

Reply:
It's almost 50:50 how the current usage is:
```console
mkilchhofer@Kicm-MBP:~/git/github/argo-helm$ git grep Env: | grep values.yaml
charts/argo-cd/values.yaml:  extraEnv: []
charts/argo-cd/values.yaml:  extraEnv: []
charts/argo-rollouts/values.yaml:  extraEnv: []
charts/argo-rollouts/values.yaml:  extraEnv: []
charts/argo-workflows/values.yaml:  extraEnv: []
charts/argo-workflows/values.yaml:  extraEnv: []
charts/argocd-image-updater/values.yaml:extraEnv: []

mkilchhofer@Kicm-MBP:~/git/github/argo-helm$ git grep env: | grep values.yaml
charts/argo-cd/ci/ha-static-values.yaml:  env:
charts/argo-cd/values.yaml:  env:
charts/argo-cd/values.yaml:  env: []
charts/argo-cd/values.yaml:  env: []
charts/argo-cd/values.yaml:  env: []
charts/argo-cd/values.yaml:  #   env:
charts/argo-cd/values.yaml:  env: []
charts/argo-events/values.yaml:  env: []
charts/argo-events/values.yaml:  env: []
charts/argo-workflows/values.yaml:  env: {}
charts/argo-workflows/values.yaml:  env: {}
charts/argocd-image-updater/values.yaml:    #   credentials: env:REGISTRY_SECRET
```

But its really inconsistent 😿 even in one chart it is different 🤣

Reply:
Lets use `env:` and `envFrom:`